client/asset/lcd/query.go

```diff
@@ -0,0 +1,33 @@
+package lcd
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/irisnet/irishub/client/context"
+	"github.com/irisnet/irishub/codec"
+)
+
+func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *codec.Codec) {
+	// Get the gateway from a moniker
+	r.HandleFunc(
+		"/asset/gateways/{moniker}",
+		monikerGatewayHandlerFn(cliCtx, cdc),
+	).Methods("GET")
+
+	// Get all gateways from an owner
+	r.HandleFunc(
+		"/asset/gateways",
+		ownerGatewaysHandlerFn(cliCtx, cdc),
+	).Methods("GET")
+}
```

Comment:
Let's keep only two query endpoints: 
- `GET /asset/gateways/{moniker}` 
- `GET /asset/gateways` which takes an optional query param: `owner`; if no param is given, then return a list of all gateways.

Reply:
Done